adafruit_platformdetect/chip.py

```diff
@@ -94,20 +97,29 @@ def _linux_id(self): # pylint: disable=too-many-branches
             if compatible and 'amlogic, g12b' in compatible:
                 linux_id = S922X
 
-        elif hardware in ("BCM2708", "BCM2709", "BCM2835"):
-            linux_id = BCM2XXX
-        elif "AM33XX" in hardware:
-            linux_id = AM33XX
-        elif "sun8i" in hardware:
-            linux_id = SUN8I
-        elif "ODROIDC" in hardware:
-            linux_id = S805
-        elif "ODROID-C2" in hardware:
-            linux_id = S905
-        elif "ODROID-N2" in hardware:
-            linux_id = S922X
-        elif "SAMA5" in hardware:
-            linux_id = SAMA5
+            # we still haven't identified the hardware, so
+            # convert it to a list and let the remaining
+            # conditions attempt.
+            if not linux_id:
+                hardware = [
+                    entry.replace('\x00', '') for entry in compatible.split(',')
+                ]
+
+        if not linux_id:
+            if set(hardware) & BCM_RANGE:
```

Comment:
Spotted my own flaw here. If `hardware` is assigned as a string from `self.detector.get_cpuinfo_field('Hardware')` above, the `set()` will iterate by character and result in things like `set(hardware) = {'b', 'c', 'm', '2', '8', '3', '5'}`. This will negate any match in `BCM_RANGE`.

Will address later today.